src/queue.rs

```diff
@@ -403,3 +409,33 @@ impl Queue {
         self.spotify.clone()
     }
 }
+
+pub fn send_notification(track_name: &str, cover_url: &str) {
+    #[cfg(feature = "cover")]
+    let res = {
+        // download album cover image
+        let path = ui::cover::cache_path_for_url(cover_url.to_string());
+
+        if !path.exists() {
+            if let Err(e) = ui::cover::download(cover_url.to_string(),
+            path.clone()) {
+                error!("Failed to download cover: {}", e);
+            }
+        }
+
+        Notification::new()
```

Comment:
I am not the developer of this project. 
I just toyed with the Idea aswell.
would it make sense to do the notification like this?
```rust
Notification::new()
            .summary(track_name)
            .body(&format!("{} - {}, album, artist))
            .icon(&path.to_str().unwrap())
            .show()
```
so the album and the artist shows up in the body of the notification.


It is actually pretty funny we both made almost the same code. (in the same code I mean both running in a new thread. your code is just so much cleaner!)

Reply:
Yeah, I think it's a good idea to include the artist as well -- fortunately, the Track::to_string function already does this: https://github.com/hrkfdn/ncspot/blob/aa8e9f10995bb71ecee78e2f7688deddbfbc0ebe/src/track.rs#L159-L163
It currently prints the artist followed by the track name in the notification. 

The current format (with this cover patch) will display a notification with the album art, artist name, and track name, exactly as the official Spotify client does. Including the album name by default might be too much. It's already a lot of text for some tracks. For example, in this example, it's already cut off at the edge of the screen with just the artist/track name.
![image](https://user-images.githubusercontent.com/1291012/125535564-dade5e63-67b6-4061-acf7-c938f09c151e.png)

I think it might be useful to be able to customize what the notification displays, but that goes beyond the scope of this PR -- maybe a follow up that adds a config for the notification format would be good...

Reply:
yeah that would make sense!

this version.
![image](https://user-images.githubusercontent.com/55736468/125599179-a770502c-79f5-43f8-b61b-55186cff0b38.png)


here is  how I have made it. Body notification shows artist and Summary show the track title.  

![image](https://user-images.githubusercontent.com/55736468/125598457-f71e6a46-1015-4430-9d67-b88050e36f9f.png)